dagshub/data_engine/model/datasource.py

```diff
@@ -742,6 +749,19 @@ def save_dataset(self, name: str) -> "Datasource":
         copy_with_ds_assigned.load_from_dataset(dataset_name=name, change_query=False)
         return copy_with_ds_assigned
 
+    def _autolog_mlflow(self):
+        if not is_mlflow_installed:
+            return
+        # Run ONLY if there's an active run going on
+        if mlflow.active_run() is None:
+            return
+        source_name = self.source.name
+        with _autolog_counter_lock:
+            counter_value = _autolog_counters[source_name]
+            _autolog_counters[source_name] += 1
```

Comment:
Couple issues here:
1. Race condition on multiprocessing, artifacts will override each other
2. If resuming an existing run and logging more to it, the counters don't persist.

Overall, it might be more sane to just use a UUID with a human readable timestamp format...
(I would stick to the timestamp formats we use in our UI when picking versions to compare, which IIRC is RFC3339)
(And some random bits before the timestamp to make it a "UUID" )